.github/workflows/tagged-releases.yml

```diff
@@ -118,5 +127,17 @@ jobs:
             GOOS=${{ matrix.goos }}
             GOARCH=${{ matrix.goarch }}
 
+      ########################### Sign built Docker image using cosign #############################
+      # https://github.blog/2021-12-06-safeguard-container-signing-capability-actions/
+      - name: Sign the published Docker image
+        if: ${{ github.event_name != 'pull_request' }}
+        env:
+          COSIGN_EXPERIMENTAL: "true"
+        # This step uses the identity token to provision an ephemeral certificate
+        # against the sigstore community Fulcio instance.
+        run: |
+          IMAGE_NAME=${{ fromJSON(steps.docker_build.outputs.metadata)['image.name'] }}
```

Comment:
```yaml
IMAGE_NAME=${{ fromJSON(steps.docker_build.outputs.metadata)['image.name'] }}
cosign sign docker.io/${{ secrets.DOCKER_HUB_ORG }}/osctrl-${{ matrix.components }}:${{ steps.vars.outputs.RELEASE_VERSION }}
```

Maybe that but I am not sure. I would have to test.

Reply:
The `cosign` command will sign every image that is pushed to DockerHub and since it is a step of the `dockerhub` job, it will just sign each image that is pushed. We could use your command but I prefer to take the output from `docker_build` because that is an image already in DockerHub.